modules/040-control-plane-manager/docs/FAQ.md

```diff
@@ -258,3 +258,36 @@ In this case, if the connection to the node is lost, the applications will be re
 Both these parameters directly impact the CPU and memory resources consumed by the control plane. By lowering timeouts, we force system components to send statuses more frequently and check the resource state more often.
 
 When deciding on the appropriate threshold values, consider resources consumed by the control nodes (graphs can help you with this). Note that the lower parameters are, the more resources you may need to allocate to these nodes.
+
+## How do make etcd backup?
+
+Login into any control-plane node with `root` user and use next script:
+
+```bash
+#!/usr/bin/env bash
+
+for node_name in $(kubectl get no -l node-role.kubernetes.io/master= -o json | jq -r '.items[].metadata.name'); do
+  kubectl -n kube-system exec "etcd-$node_name" -- /bin/sh -c 'ETCDCTL_API=3 /usr/bin/etcdctl --cacert /etc/kubernetes/pki/etcd/ca.crt --cert /etc/kubernetes/pki/etcd/ca.crt --key /etc/kubernetes/pki/etcd/ca.key --endpoints https://127.0.0.1:2379/ snapshot save /tmp/etcd-backup' && \
+  kubectl -n kube-system exec "etcd-$node_name" -- sha256sum /tmp/etcd-backup | cut -f 1 -d" " | xargs echo -n > etc-backup.snapshot.shasum && \
+  kubectl -n kube-system exec "etcd-$node_name" -- cat /tmp/etcd-backup > etc-backup.snapshot && \
```

Comment:
However both commands are not working to me 🤔

Reply:
> kubectl has built-in cp command

Yep i know, but cp has one problem: with large file cp can often get error `error: unexpected EOF` because `cp` does not reconnect (in 1.23 it will be fixed with [flag](https://github.com/kubernetes/kubernetes/pull/104792)) but `exec` command  reconnect always.

Reply:
> Also this command will copy snapshot from all the nodes and override it, forgot $node_name, in output file?

No. We want get one snapshot file. We use loop over all etcd pod because one pod can be unavailable.

Reply:
> Yep i know, but cp has one problem: with large file cp can often get error error: unexpected EOF because cp does not reconnect (in 1.23 it will be fixed with flag) but exec command always reconnect.

Interesting, I didn't know about that. Any way both commands were not working to me.

> No. We want get one snapshot file. We use loop over all etcd pod because one pod can be unavailable.

This way suggested script will not be finished after fist successful transfer, but will continue backing up other pods with overwriting the first file

Reply:
i've got your [script](https://github.com/deckhouse/deckhouse/pull/1908#discussion_r913510044) and i've added `break`.